### PR TITLE
normalize datasets via DataStream.default_stream

### LINKED ISSUE #15: Normalize tfd and mnist datasets
The tfd and mnist datasets return data between 0 and 255.

We need to normalize them to 0..1 before feeding them into the binary crossentropy.

### PR DESCRIPTION
**Pull request by dribnet**
Fixes #15 

Calling class method `DataStream.default_stream` [applies the default transforms](https://github.com/mila-udem/fuel/blob/eb1e7bdca9a0ab46bfc2b728706fa06cb24f14cd/fuel/streams.py#L145) on the datastream. The base Dataset class then calls the [set of default_transforms](https://github.com/mila-udem/fuel/blob/eb1e7bdca9a0ab46bfc2b728706fa06cb24f14cd/fuel/datasets/base.py#L84) which can be specific to each subclass. The [default_transform for mnist](https://github.com/mila-udem/fuel/blob/eb1e7bdca9a0ab46bfc2b728706fa06cb24f14cd/fuel/datasets/mnist.py#L31) normalizes the pixels to the [0 1] range via `uint8_pixels_to_floatX(('features',))`

In other words: fuel seems to have a mechanism to do exactly what we need - call `DataStream.default_stream` and the incoming data will be normalized in a dataset specific way. I confirmed that this seems to be working with mnist and doesn't adversely affect bmnist. (I didn't test against tfd)

Note also that this change is a single commit (2550fc6) downstream from commits in #18

### CODE REVIEW

**Comment by jbornschein** [comment]
Perfect!
